### DISCUSSION TITLE
Calling functions from base class in template class compiles with gcc but not clang

### DISCUSSION THREAD

**Discussion by smpark7** [idaholab/moose#27738] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

I recently created a template material class [SATauMaterial](https://github.com/arfc/moltres/blob/devel/include/materials/SATauMaterial.h) in the Moltres app that primarily inherits from another template material class [INSADTauMaterial](https://github.com/idaholab/moose/blob/next/modules/navier_stokes/include/materials/INSADTauMaterial.h).

The app compiles without errors on Linux machines, but encounters the following error on MacOS machines:
```
error: 'adCoupledValue' following the 'template' keyword does not refer to a template
    _mu_tilde(this->template adCoupledValue("mu_tilde")),
```
for `adCoupledValue` and a couple of other functions prefixed with `this->template`. [Full error log](https://github.com/arfc/moltres/files/15474761/errors.txt)

I could only get as far as identifying that this error arises from differences in supported features between gcc and clang. I appreciate any help with this compile error.

**Comment by GiudGiud** [comment]
Is your new class a template?

Why are you using this->template for these calls?
They are not templates so you can just call the routine directly with no "this->template"

**Reply by smpark7** [reply]
Yes it is also a template `SATauMaterialTempl`

**Reply by GiudGiud** [reply, marked as answer]
I thought this->template was for using templated routines within a templated class. I may be wrong. 

If I am not, then adCoupledValue is a not a template so no need for these extra qualifiers

**Reply by smpark7** [reply]
You are right. I removed this->template for non-template calls and added using Coupleable::... Thanks for your help!
